src/cnxml/se.tsx

```diff
@@ -0,0 +1,453 @@
+// Copyright 2020 OpenStax Poland
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license text.
+
+/** @jsx JSX.createElement */
+
+import { Editor, Element, Node, Text } from 'slate'
+
+import * as JSX from './jsx'
+import {
+    Admonition, AltText, Audio, Caption, Code, Commentary, CrossReference,
+    Definition, DefinitionExample, DefinitionTerm, DocumentReference, Exercise,
+    Figure, Footnote, Foreign, Glossary, Image, Link, List, ListItem, Meaning,
+    Media, MediaData, Paragraph, Preformat, Problem, Proof, Quotation, Rule,
+    RuleExample, Section, SeeAlso, Solution, Statement, StyledText, Term, Title,
+    Video, WithClasses,
+} from '../interfaces'
+import { uuid } from '../util'
+import { Document as Doc } from '.'
+
+/** Serialization options */
+export interface Options<Format> {
+    /** Format to which to serialize */
+    format: Format
+    /** Function used to determine the MIME type of a media item */
+    mediaMime: MediaMimeFunction
+}
+
+/** Function used to determine the MIME type of a media item */
+export type MediaMimeFunction = (media: MediaData) => string
+
+export default function serialize(
+    editor: Editor, document: Doc, options: Options<'xml'>): string
+export default function serialize(
+    editor: Editor, document: Doc, options: Options<'dom'>): Document
+
+/** Serialize a document to CNXML */
+export default function serialize(
+    editor: Editor,
+    doc: Doc,
+    options: Options<'xml' | 'dom'>,
+): string | Document {
+    const { format, mediaMime } = options
+
+    const content = doc.content.map(n => serializeNode(editor, n, { mediaMime }))
+    const glossary = Glossary.isGlossary(doc.content[doc.content.length - 1])
+        ? content.pop()
+        : null
+
+    const document = JSX.render(<document
+        xmlns={JSX.CNXML_NAMESPACE}
+        cnxml-version={doc.version}
+        id={doc.moduleId}
+        module-id={doc.moduleId}
+        >
+        <title>{doc.title}</title>
+        <content>
+            {content}
+        </content>
+        {glossary}
+    </document>)
+
+    if (format === 'dom') {
+        return document
+    } else {
+        let x = new XMLSerializer().serializeToString(document)
+
+        if (!x.startsWith('<?xml')) {
+            x = `<?xml version="1.0" encoding="${document.characterSet}">\n${x}`
+        }
+
+        return x
+    }
+}
+
+/** Serialization context */
+interface Context {
+    mediaMime: MediaMimeFunction
+}
+
+/** Attributes common to all elements */
+interface CommonAttrs {
+    /** Element's ID */
+    id: string
+    /** CSS classes */
+    class?: string
+}
+
+/** Serialize a single node */
+function serializeNode(editor: Editor, node: Node, ctx: Context): JSX.Node {
+    if (Text.isText(node)) {
+        return node.text
+    }
```

Comment:
`StyledText` will also return `true` in this case, right? If yes then should we check for `isPlainText` instead here and handle marks globally?